pkg/actuators/machine/instances.go

```diff
@@ -422,9 +427,28 @@ func buildTagList(machineName string, clusterID string, machineTags []awsprovide
 		{Key: aws.String("kubernetes.io/cluster/" + clusterID), Value: aws.String("owned")},
 		{Key: aws.String("Name"), Value: aws.String(machineName)},
 	}...)
+
 	return removeDuplicatedTags(rawTagList)
 }
 
+// mergeInfrastructureAndMachineSpecTags merge list of tags from machine provider spec and Infrastructure object platform spec.
+// Machine tags have precedence over Infrastructure
+func mergeInfrastructureAndMachineSpecTags(machineSpecTags []awsproviderv1.TagSpecification, infra *configv1.Infrastructure) []awsproviderv1.TagSpecification {
+	if infra == nil || infra.Status.PlatformStatus == nil || infra.Status.PlatformStatus.AWS == nil || infra.Status.PlatformStatus.AWS.ResourceTags == nil {
```

Comment:
Check for nil infra.Status here as well?  I don't think that status is always guaranteed to be present.

Reply:
It's not a pointer so I don't think it's required https://github.com/openshift/api/blob/a3c8fa53e01bc4442b687d3aca1571692f7676ce/config/v1/types_infrastructure.go#L11-L22